src/hooks/useTokenAllowance.ts

```diff
@@ -15,3 +18,34 @@ export function useTokenAllowance(token?: Token, owner?: string, spender?: strin
     [token, allowance]
   )
 }
+
+export function useUpdateTokenAllowance(amount?: CurrencyAmount<Token>, spender?: string) {
+  const contract = useTokenContract(amount?.currency.address)
+
+  return useCallback(async (): Promise<ApprovalTransactionInfo> => {
+    try {
+      if (!amount) throw new Error('missing amount')
+      if (!contract) throw new Error('missing contract')
+      if (!spender) throw new Error('missing spender')
+
+      let allowance: BigNumberish = MaxUint256.toString()
```

Comment:
why does this need to be in string form?

Reply:
conversion between large number representations